docs/api.yaml

```diff
@@ -0,0 +1,925 @@
+              example: 0.1
+            wait_seconds:
+              type: integer
+              description: "wait in seconds before attempting next increase/decrease step."
+              example: 10
+            ping_seconds:
+              type: integer
+              description: "When fully blocked, keep checking every ping_seconds for recovery."
+              example: 20
+            stop_after_seconds:
+              type: integer
+              description: "Mark consumption stopped if recovery doesn't happen till this time. Manual intervention needed post this."
+              example: 7200
+        
+    topic:
+      type: object
+      properties:
+        name:
+          type: string
+          example: "user-invoices"
+        tenant:
+          type: string
+          example: "all-invoices"
+        grouped:
+          type: boolean
+          example: true
+        max_subscriptions:
+          type: integer
+          minimum: 1
+          example: 1
+        properties:
+          $ref: '#/components/schemas/property_bag'
+        retention_policy:
+          $ref: '#/components/schemas/retention_policy'
+        capacity_policy:
+          $ref: '#/components/schemas/capacity_policy'
+        state:
+          $ref: '#/components/schemas/state_enum'
+        version:
+          type: integer
+          example: 1
+          
+    topics:
+      type: array
+      items:
+        $ref: '#/components/schemas/topic'
+          
+    endpoint:
+      type: object
+      properties:
+        url:
+          type: string
+          example: "https://invoice-signer-service/invoice/sign"
+        method:
+          type: string
+          example: "POST"
+          enum: 
+            - "POST"
+            - "GET"
+            - "PUT"
+        connect_timeout:
+          type: integer
+          example: 4
+          default: 5
+        request_timeout:
+          type: integer
+          example: 4
+          default: 5
+        auth_info:
+          type: object
+          properties:
+            audience:
+              type: string
+              example: "invoice-signer-service"
+          nullable: true
+          
+    subscription:
+      type: object
+      properties:
+        name:
+          type: string
+          example: "my-sub"
+        tenant:
+          type: string
+          example: "all-invoices"
+        topic:
+          type: string
+          example: "user-invoices"
+        grouped:
+          type: boolean
+          example: true
+        endpoint:
+          $ref: '#/components/schemas/endpoint'
+        retention_policy:
+          $ref: '#/components/schemas/retention_policy'
+        retry_policy:
+          $ref: '#/components/schemas/retry_policy'
+        dlq_policy:
+          $ref: '#/components/schemas/dlq_policy'
+        consumption_policy:
+          $ref: '#/components/schemas/consumption_policy'
+        version:
```

Comment:
when we say version, we mean schema version right?

Reply:
This was object versioning and not schema versioning.
Shall we cover schema version using API versioning ?